### DISCUSSION TITLE
[Chapter 11] Spring 스케줄과 배치의 예시

### DISCUSSION THREAD

**Discussion by ahngilwoong** [cowaine/spring-boot-for-msa#182] [Chapter 11. Spring Scheduling Tasks] [answered]
이번내용은 스케줄을 써봐야 실질적으로 공감이 가는 챕터 인 것 같습니다.
배치를 한번도 안써본 입장에서 질문 남깁니다.
현재 책에는 스케줄에 대해서만 내용이 있는데 스케줄과 배치를 연동해서 사용하는 예시를 간략하게 설명해주실 분이 있다면 감사히 설명을 듣겠습니다.
만약 매일 은행에서 밤마다 점검시간에 스케줄로 배치를 돌려 일괄적으로 작업을 처리한다.
여기서 자세히 잡 ? 스텝? 아이템 리더? 프로세서? 라는 용어를 비유해서 설명해주신다면 정말 감사하겠습니다.

**Comment by JoisFe** [comment, marked as answer]
스프링 배치에 대해서 얘기하는 것 같은데
이번 책에서는 스프링 배치 기술 보다는 배치성 작업을 위한 스프링 스케줄링 기능에 대한 내용이 주 인 것 같습니다.

스프링 배치는 Job을 구동하거나 실행하는게 아니라 위에서 언급한 스케줄링 등의 기능을 이용하여 Job을 구동합니다. 스프링 배치는 대용량 데이터에 대해 일괄 처리를 위한 프레임워크로 스케줄링 기능 혹은 Jenkins, Quartz 등으로 Job을 구동할 때 트랜잭션 처리, 벌크성 고성능 작업 등을 위해 여러 서비스 및 기능을 제공하는 프레임워크 입니다.

Spring Batch에서  Job, Step, ItemReader, ItemProcessor, ItemWriter 경우
Spring Doc에 매우 상세히 설명을 해주어서 해당 링크를 남깁니다.
- [Job](https://docs.spring.io/spring-batch/reference/job.html)
- [Step](https://docs.spring.io/spring-batch/reference/step.html)
- [ItemReader](
https://docs.spring.io/spring-batch/reference/readers-and-writers/item-reader.html)
- [ItemWriter](https://docs.spring.io/spring-batch/reference/readers-and-writers/item-writer.html)
- [ItemProcessor](https://docs.spring.io/spring-batch/reference/processor.html)

Chunk 등 다양한 여러 개념 또한 알아보시는 걸 추천합니다.!

**Comment by Irisation23** [comment]
### 짧 정리

1. 물류 시스템에 재고를 조회하는 파트가 있음 -> 창고 시스템과 외부 실 재고간의 데이터 간극이 생길 수 있음 -> 이를 5분 단위로 처리하는 마이크로 배치가 있음

2. 주문에 대한 모니터링 시스템이 있음 -> 만약 주문에 데이터 정체가 생기거나 오류가 생긴다면?  해당 모니터링 툴(e.g 키바나) 등등의 데이터를 참조해서 작업을 처리함 -> 이때 30분에 한번 or 10분에 한번씩 에러를 잡아 Slack 메시지를 보냄
